public/app/plugins/datasource/loki/components/useLokiLabels.ts

```diff
@@ -66,14 +75,12 @@ export const useLokiLabels = (
   // Since shouldTryRefreshLabels is reset AFTER the labels are refreshed we are secured in case of trying to refresh
   // when previous refresh hasn't finished yet
   useEffect(() => {
-    if (shouldTryRefreshLabels) {
-      tryLabelsRefresh();
-    }
+    tryLabelsRefresh(shouldTryRefreshLabels);
   }, [shouldTryRefreshLabels]);
```

Comment:
Hm, getting rid of `useLokiForceRefresh` hook and moving the datasourceStatus could simplify this hook's logic. If, instead of introducing this:

```ts
const [shouldTryRefreshLabels, setRefreshLabels] = useState({ refresh: false, force: false });
```

you added:
```ts
const [shouldForceRefreshLabels, setForceRefreshLabels] = useState(false)
```

then this effect could be responsive to `dataSourceStatus` argument changes as well setting the `shouldForceRefreshLabels` to true if the status === connected. And then, in `tryLabelsRefresh`, when it's eventually executed on labels selector open event, it would simply pass the `shouldForceRefreshLabels` to languageProvider.refreshLogLabels and set it to `false` when labels are fetched. And the exposed `refreshLabels` wouldn't have to be changed below, I think

Reply:
I see what you mean, but I still think we need the previous state otherwise we'll get a force refresh for the initial state. I'll move the force refresh logic into this instead.